runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -2419,6 +2466,18 @@ ClassFileOracle::methodIsNonStaticNonAbstract(U_16 methodIndex)
 	return J9_ARE_NO_BITS_SET(_classFile->methods[methodIndex].accessFlags, (CFR_ACC_STATIC | CFR_ACC_ABSTRACT));
 }
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+bool
+ClassFileOracle::methodIsSynchronized(U_16 methodIndex)
+{
+	ROMCLASS_VERBOSE_PHASE_HOT(_context, MethodIsSynchronized);
```

Comment:
We should also double check that if this check applies to static methods as well

Reply:
You mean if this check will capture static synchronized method 
or whether we want to inject IdentityInterface if there is only static synchronized method ?

The code here will capture static synchronized method, so IdentityInterface will be injected if there is a synchronized method (no matter it is static or not). I don't see any distinction between static vs non-static synchronized method in the description of #12039

Reply:
> whether we want to inject IdentityInterface if there is only static synchronized method ?

This.

Can you confirm with RI that this is the behaviour? Instance synch methods are disallowed for obvious reasons, but static synch could be allowed since synchronization is on the j.l.Class.

Reply:
RI allows static synchronized method. Also it allows static fields.